### PR TITLE
Spec cleanup

### PR DESCRIPTION
**Pull request by timabell**
Prep work for incremental fetch paging.

This is a big diff so sending this off to master to avoid future rebase hell.

Pure refactor

### CODE REVIEW

File: spec/requests/api/v1/provider_spec.rb [hunk 1]
```diff
@@ -66,156 +11,216 @@
...
-      get '/api/v1/providers', headers: { 'HTTP_AUTHORIZATION' => credentials }
-      expect(response).to have_http_status(:success)
-    end
-
-    it 'returns http unauthorised' do
-      get '/api/v1/providers',
-          headers: { 'HTTP_AUTHORIZATION' => unauthorized_credentials }
-      expect(response).to have_http_status(:unauthorized)
-    end
+    describe "without changed_since parameter" do
```
Comment (misaka) [inline spec/requests/api/v1/provider_spec.rb]: I think this is a `context`.

File: spec/requests/api/v1/provider_spec.rb [hunk 2]
```diff
@@ -66,156 +11,216 @@
...
+                              ])
+        end
       end
     end
-    context 'with changed_since' do
-      it 'JSON body response contains expected providers' do
-        old_provider = create(:provider, provider_code: "SINCE1", age: 1.hour.ago)
 
-        updated_provider = create(:provider, provider_code: "SINCE2", age: 5.minutes.ago)
+    describe "with changed_since parameter" do
```
Comment (misaka) [inline spec/requests/api/v1/provider_spec.rb]: And I think this should be a `context` too.

**Comment by dankmitchell** [comment]
Can we squash the commits, please?

**Comment by timabell** [comment]
[Implement incremental fetch for providers.](https://trello.com/c/HMAlga4l/852-implement-incremental-fetch-for-providers)

**Pull request merged**